### ISSUE TITLE
Parallax 2.4r2 Not Installing True Type Font

### ISSUE THREAD

**Issue by cjmoorejr** [parallaxinc/Propeller-Tool#46]

**Comment by cjmoorejr** [comment]
The Version 2.4r2 installer does NOT install the Parallax (Regular) True Type Font and the resulting text in the editor has weird spacing problems.  If you have version 1.3.2 installed (which does install the font) everything will look OK.  Unfortunately, for new users the new website does not have version 1.3.2.  They could navigate to the old site to get it though.

**Comment by PropGit** [comment]
Thanks @cjmoorejr.

The font is included in the package and I see that it's placed in the same folder as the editor.  In the rush to rebuild the installer, I was unable to figure out how to get the installer to "install" the font, so I relied on the Propeller Tool's ability to automatically do a session-install of the font if it detects the font isn't active in the OS.  I tested this on a virgin machine and it seemed to work (I'll retest); however, I have seen in the past that this mysteriously fails to work sometimes.

Regardless, thank you for reporting this as I think I've now figured out how to get new installer to do it and will attempt it today.

**Comment by PropGit** [comment]
Fixed in v2.4 (r3).  https://www.parallax.com/package/propeller-tool-software-for-windows-spin-assembly-2/

Tested on virgin Win 10 VM; Parallax.ttf is installed to Windows/fonts folder and appears proper when Propeller Tool is run.

**Comment by cjmoorejr** [comment]
Just tested 2.4r3 on my system and the font did not install.  Tried it twice.  The second time after a reboot.  I was able to drag the font into the windows/font directory and it worked.  I'm including my particulars on my system.

Microsoft Surface Pro 7
Edition:  Windows 10 Home
Version:  20H2
OS Build:  19042.630

I installed on D: instead of C:

Maybe the install script is trying to copy the font from the C: drive instead of the drive (D:) I specified at the beginning of the install process.

**Comment by PropGit** [comment]
Thanks @cjmoorejr.  It's possible the installation drive has something to do with it, but the install script is specifying the installer source as the source for the font install (wherever it's being run from) and the system font folder (by category, not by folder path), so I'm surprised this is happening.  I don't have a way to duplicate with a separate drive at the moment, but I'll try to do so to prove/disprove our install destination theory.

If you're inclined to try again, by first manually uninstalling the font, then uninstalling Propeller Tool, we may be able to log the issue at fault to find the source of the problem.  This can be done via command line or by creating a shortcut to your Setup-Propeller-Tool-v2.4r3.exe, then right-click that shortcut and set the target to be something like this:
C:\Downloads\Setup-Propeller-Tool_v2.4r3.exe  /LOG="C:\FileName.log"

That will generate a log file that we can review for "font" to see if it reported an error of some kind during installation.

**Comment by PropGit** [comment]
I just ran this test on my side with the log file enabled, then searched for "font" in the generated log file.  This is what mine looks like:
```
2020-11-24 14:56:32.309   -- File entry --
2020-11-24 14:56:32.309   Dest filename: C:\WINDOWS\Fonts\Parallax.ttf
2020-11-24 14:56:32.368   Time stamp of our file: 2020-11-24 08:28:06.000
2020-11-24 14:56:32.368   Installing the file.
2020-11-24 14:56:32.496   Successfully installed the file.
2020-11-24 14:56:32.496   Registering file as a font ("Parallax (TrueType)")
```

**Comment by cjmoorejr** [comment]
Eureka!  I've attached two files.  After uninstalling and reinstalling both 2.4r3 and 1.3.2 several times the short answers is:

1.3.2 installs the following Parallax.ttf file:

-a----         1/25/2006   8:34 AM        1444060 Parallax.ttf

which is NOT removed when you uninstall 1.3.2 even though the /windows/font explorer window DOESN'T show the font.  You have to remove it via command line (in my case PowerShell).  Once this is done, the 2.4r3 installation is successful (as Filename2.log shows).  Not sure why the 1.3.2 uninstaller leaves the file behind but that's a mystery for another day (i.e. issue) which is probably mute as this point.  That's for the LOG suggestion.  You learn something new everyday.  Sorry it took me so long to get back to you on this and for wasting your time.

[Filename.log](https://github.com/parallaxinc/Propeller-Tool/files/5594326/Filename.log)
[parallax.txt](https://github.com/parallaxinc/Propeller-Tool/files/5594327/parallax.txt)

[Filename2.log](https://github.com/parallaxinc/Propeller-Tool/files/5594356/Filename2.log)

**Comment by PropGit** [comment]
Thanks @cjmoorejr!  Yes, that would do it, and I'm thinking since I changed installer systems, the new one may be only looking for file existence based on what it has installed on that system in prior times.  I'll retry the steps you followed on my system as soon as I can.

**Comment by PropGit** [comment]
Thank you @cjmoorejr - I found the same to be true on my system... the Parallax font is there, but seems hidden from the visual Explorer view.  This is what happens with a font is copied to the folder (manually somehow) without actually "installing" it for the system... it, the file exists, but the font in the file was never registered with the system.  Windows Explorer automatically registers fonts that are copied to the fonts folder, but copying a file via command-line or via other method does not perform the registration.

Regardless, v1.3.2 isn't uninstalling it properly.

I've fixed this in the v2.4.1 installer, which installs the font whether or not it sees that the font was already installed.
